tensorflow_probability/python/distributions/loglogistic.py

```diff
@@ -0,0 +1,161 @@
+
+    The LogLogistic distribution models positive-valued random variables
+    whose logarithm is a logistic distribution with loc `loc` and
+    scale `scale`. It is constructed as the exponential
+    transformation of a Logistic distribution.
+
+    Args:
+      loc: Floating-point `Tensor`;
+        the loc of the underlying logistic distribution(s).
+      scale: Floating-point `Tensor`;
+        the scale of the underlying logistic distribution(s).
+      validate_args: Python `bool`, default `False`. Whether to validate input
+        with asserts. If `validate_args` is `False`, and the inputs are
+        invalid, correct behavior is not guaranteed.
+      allow_nan_stats: Python `bool`, default `True`. If `False`, raise an
+        exception if a statistic (e.g. mean/mode/etc...) is undefined for any
+        batch member If `True`, batch members with valid parameters leading to
+        undefined statistics will return NaN for this statistic.
+      name: The name to give Ops created by the initializer.
+    """
+    parameters = dict(locals())
+    with tf.name_scope(name) as name:
+      super(LogLogistic, self).__init__(
+          distribution=logistic.Logistic(
+            loc=loc,
+            scale=scale,
+            allow_nan_stats=allow_nan_stats
+          ),
+          bijector=exp_bijector.Exp(),
+          validate_args=validate_args,
+          parameters=parameters,
+          name=name)
+
+  @classmethod
+  def _params_event_ndims(cls):
+    return dict(scale=0, concentration=0)
+
+  @property
+  def loc(self):
+    """Distribution parameter of the underlying distribution."""
+    return self.distribution.loc
+
+  @property
+  def scale(self):
+    """Distribution parameter of the underlying distribution"""
+    return self.distribution.scale
+
+  def _mean(self):
+    scale = tf.convert_to_tensor(self.scale)
+    with tf.control_dependencies(
+        [] if self.allow_nan_stats else [  # pylint: disable=g-long-ternary
+          assert_util.assert_less(
+            scale,
+            tf.ones([], dtype=self.dtype),
+            message='Mean undefined for scale > 1.'),
+        ]):
+      mean = tf.math.exp(self.loc) / sinc(scale)
+      return tf.where(scale > 1., np.nan, mean)
+
+  def _variance(self):
+    scale = tf.convert_to_tensor(self.scale)
+    with tf.control_dependencies(
+    [] if self.allow_nan_stats else [  # pylint: disable=g-long-ternary
+          assert_util.assert_less(
+            scale,
+            0.5 * tf.ones([], dtype=self.dtype),
+            message='Variance undefined for scale > 1/2.'),
+        ]):
+      variance = tf.math.exp(2 * self.loc) * (
+          1. / sinc(2 * scale) - 1. / sinc(scale) ** 2)
+      return tf.where(scale > 0.5, np.nan, variance)
+
+  def _mode(self):
+    log_mode = self.loc + self.scale * (
+        tf.math.log1p(- self.scale) - tf.math.log1p(self.scale)
+    )
+    return tf.where(self.scale > 1., 0., tf.math.exp(log_mode))
```

Comment:
You'll need to cast `0.` and `np.nan` in these wheres unfortunately (this is for dtype=tf.float64).